src/conn/conn_tiered.c

```diff
@@ -749,15 +741,10 @@ __wt_tiered_storage_create(WT_SESSION_IMPL *session, const char *cfg[])
 
     /* Start the internal thread. */
     WT_ERR(__wt_cond_alloc(session, "flush tier", &conn->flush_cond));
-    WT_ERR(__wt_cond_alloc(session, "storage server", &conn->tiered_cond));
-    FLD_SET(conn->server_flags, WT_CONN_SERVER_TIERED);
-
-    WT_ERR(__wt_open_internal_session(conn, "storage-server", true, 0, 0, &conn->tiered_session));
-    session = conn->tiered_session;
 
-    /* Start the thread. */
-    WT_ERR(__wt_thread_create(session, &conn->tiered_tid, __tiered_server, session));
-    conn->tiered_tid_set = true;
+    WT_ERR(__wt_thread_start(conn, "storage-server", true, 0, "storage server", 0, 0,
+      __tiered_server, &conn->tiered_thread));
+    FLD_SET(conn->server_flags, WT_CONN_SERVER_TIERED);
```

Comment:
With this change the threads internal session will now be opened before WT_CONN_SERVER_TIERED is set.
I don't expect this to cause issues, wanted to identify it to be safe.

Reply:
This is an important flagging. Why do you need that change? It will cause issues. If the thread is started and runs in `tiered_server` before this thread comes back and sets that flag and the new thread sees it in its view of the memory, the new thread will exit. There are a lot of places that use `WT_CONN_SERVER_<whatever>` in this same way. For example, you didn't need to do this with `conn_capacity.c` and setting `WT_CONN_SERVER_CAPACITY`.

Reply:
I initially did this to have the flag set after the conditions are allocated as it was in the original code, but I'll fix it to make sure the flag is set before `thread_start` instead. Thanks!

Reply:
Fixed in 4fd985f